### ISSUE TITLE
How to convolve two tensor according to the batch axis?

### ISSUE THREAD

**Issue by bugzhu** [tensorflow/tensorflow#34570]
In my problem, I want to convolve two tensors in my neural network model. The shape of two tensors is [None, 2, 1], [None, 3, 1] respectively. The axis with dimension None means the batch size of the input tensor. For each sample in batch, I want to convolve the two tensors with shape [2, 1] and [3, 1].
However, the tf.nn.conv1d in TensorFlow can only convolve the input with a fixed kernel. Is there any function that can support the convolution of two tensors according to the batch size axis, similar to the tf.multiply which can element wisely multiply two tensors.
